### PR TITLE
ferretdb: init at 0.5.2

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes

FerretDB (previously MangoDB) was founded to become the de-facto open-source substitute to MongoDB. FerretDB is an open-source proxy, converting the MongoDB 5.0+ wire protocol queries to SQL - using PostgreSQL as a database engine.

https://github.com/FerretDB/FerretDB

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 188584` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ferretdb</li>
  </ul>
</details>

**Review by azahi — APPROVED** [review]

**Review by NetaliDev — APPROVED** [review]
Builds and seems to run.

File: pkgs/servers/nosql/ferretdb/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, git
+, tree
```
Comment (bobby285271) [inline pkgs/servers/nosql/ferretdb/default.nix:5]: Probably not used :thinking:

Comment (dit7ya) [reply pkgs/servers/nosql/ferretdb/default.nix:5]: You are right :thinking: it is leftover from debugging.

**Comment by risicle** [comment]
Please squash your commits.

File: pkgs/servers/nosql/ferretdb/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  CGO_ENABLED = 0;
+
+  subPackages = [ "cmd/ferretdb" ];
+
+  # We cannot run the `go generate` command since it requires full git history
+  # (deepClone and leaveDotGit does not seem to help)
+  preBuild = ''
+    echo ${version} > internal/util/version/gen/version.txt
+  '';
+
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/ferretdb/default.nix:27]:
```suggestion
```

File: pkgs/servers/nosql/ferretdb/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  pname = "FerretDB";
+  version = "0.5.2";
+
+  src = fetchFromGitHub {
+    owner = "FerretDB";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-WSdscZ1/Dix83RE95Iv61rdaSBWx1GMi6qOIPNus+ZI=";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/ferretdb/default.nix:15]:
```suggestion

  postPatch = ''
    echo ${version} > internal/util/version/gen/version.txt
  '';

```

File: pkgs/servers/nosql/ferretdb/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+, buildGoModule
+, fetchFromGitHub
+}:
+buildGoModule rec {
+  pname = "FerretDB";
+  version = "0.5.2";
+
+  src = fetchFromGitHub {
+    owner = "FerretDB";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/servers/nosql/ferretdb/default.nix:11]:
```suggestion
}:

buildGoModule rec {
  pname = "ferretdb";
  version = "0.5.2";

  src = fetchFromGitHub {
    owner = "FerretDB";
    repo = "FerretDB";
```

**Pull request merged**